core/src/main/java/com/cloud/agent/api/VmStatsEntry.java

```diff
@@ -20,151 +20,17 @@
 package com.cloud.agent.api;
 
 import com.cloud.vm.UserVmVO;
-import com.cloud.vm.VmStats;
 
-public class VmStatsEntry implements VmStats {
+public class VmStatsEntry extends VmStatsEntryBase {
```

Comment:
Any added advantage of adding the new VmStatsEntryBase class?

Reply:
@ravening yes
`VmStatsEntryBase` class represents the VM metrics. When persisting a stats collection, this class is converted to JSON and assigned to the `vm_stats_data` field of the `vm_stats` table. `VmStatsEntry` class (which extends `VmStatsEntryBase`) has an `UserVmVO` attribute that is used in another piece of code, but does not need to be included in a `vm_stats` table record.